06-building-a-program-start-to-finish/solving-the-problem/output/app/class/invertedIndexClass.js

```diff
@@ -0,0 +1,199 @@
+   * @param {Array} fileName
+   * @param {string} file
+   * @returns {boolean} returns a boolean
+  */
+  createIndex(fileName, file) {
+    if (!file) { return false; }
+
+    const words = [];
+    const indexedWords = {};
+    file.forEach((book) => {
+      words.push(InvertedIndex.tokenize(book));
+    });
+
+    words.forEach((book, index) => {
+      for (let i = 0; i < book.length; i += 1) {
+        if (!Object.prototype.hasOwnProperty.call(indexedWords, book[i])) {
+          indexedWords[book[i]] = [];
+        }
+        indexedWords[book[i]].push(index);
+      }
+    });
+
+    if (!Object.prototype.hasOwnProperty.call(this.indices, fileName)) {
+      this.indices[fileName] = indexedWords;
+    }
+    this.indexedFiles[fileName] = file.length;
+    return true;
+  }
+
+  /**
+   * function to fetch generated indices
+   * @function
+   * @param {string} fileName
+   * @returns {Object} indices
+  */
+  getIndices(fileName) {
+    return typeof fileName === 'undefined'
+    || typeof fileName !== 'string' ? this.indices : this.indices[fileName];
+  }
+
+  /**
+   * function to search through created indices
+   * @function
+  * @param {String} fileName uploaded valid JSON file
+  * @param {String} query word(s) or terms to search for
+  * @returns {array} returns array searchResults contained search indices
+  */
+  searchIndices(fileName, query) {
+    const searchResults = [];
+    const result = {};
+    let indices = {};
+
+    if (this.getIndices(fileName)) {
+      indices[fileName] = this.getIndices(fileName);
+    } else {
+      indices = this.indices;
+    }
+
+    Object.keys(indices).forEach((book) => {
+      query.split(' ').forEach((word) => {
+        if (Object.prototype.hasOwnProperty.call(indices[book], word)) {
+          if (!Object.prototype.hasOwnProperty.call(result, book)) {
+            result[book] = {
+              terms: {},
+              fileName: book
+            };
+          }
+          result[book].terms[word] = indices[book][word];
+        }
+      });
+
+      searchResults.push(result[book]);
+    });
+
+    if (typeof searchResults[0] === 'undefined') {
+      return false;
+    }
+    return searchResults;
+  }
+
+
+  /**
+   * function to validate user uploaded files
+   * @function
+   * @param {object} file
+   * @returns {boolean} return boolean for all test cases
+  */
+  static validateFile(file) {
+    let isValid = file;
+
+    if (file.length > 0) {
+      for (let i = 0; i < file.length; i += 1) {
+        if (file[i] === undefined
+          || file[i].title === '' || file[i].text === ''
+          || !file[i].title || !file[i].text
+          || typeof file[i].text !== 'string') {
+          isValid = false;
+        }
+      }
+    } else isValid = false;
+
+    return isValid;
+  }
+
+  /**
+   * remove duplicates from word array
+   * @function
+   * @param {array} array
+   * @returns {array} array equivalent with unique items
+  */
+  static getUnique(array) {
```

Comment:
Is array a dataType or your parameter name? I believe you can use something better as a parameter name.